netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -234,9 +244,43 @@ private void onHeadersRead(int streamId, Http2Headers headers, boolean endStream
   /**
    * Handler for an inbound HTTP/2 DATA frame.
    */
-  private void onDataRead(int streamId, ByteBuf data, boolean endOfStream) {
+
+  private void onDataRead(int streamId, ByteBuf data, int padding, boolean endOfStream) {
+    dataSizeSincePing += data.readableBytes() + padding;
     NettyClientStream stream = clientStream(requireHttp2Stream(streamId));
     stream.transportDataReceived(data, endOfStream);
+    if (!pinging) {
```

Comment:
I'd feel better if this block happened before the dataSizeSincePing increment (so either move the increment down or the block up) to avoid issues with small window sizes. I could be convinced otherwise though.